include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -0,0 +1,310 @@
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the emble file.
+    sequence_file_format_embl(sequence_file_format_embl const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the emble file.
+    sequence_file_format_embl & operator=(sequence_file_format_embl const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_format_embl(sequence_file_format_embl &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_format_embl & operator=(sequence_file_format_embl &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "embl" },
+    };
+
+    //!\copydoc sequence_file_input_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                            & stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                                                               & sequence,
+              id_type                                                                & id,
+              qual_type                                                              & SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+        auto stream_it = ranges::begin(stream_view);
+
+        std::string identifier;
+        ranges::copy(stream_view | view::take_until_or_throw(is_cntrl || is_blank),
+                ranges::back_insert_iterator{identifier});
+        // Jump over header
+        while (identifier != "ID")
+        {
+            detail::consume(stream_view | view::take_until(is_cntrl));
+            ++stream_it;
+            identifier = "";
+            ranges::copy(stream_view | view::take_until_or_throw(is_cntrl || is_blank),
+                    ranges::back_insert_iterator{identifier});
+        }
+
+        // ID
+        detail::consume(stream_view | ranges::view::take_while(is_blank));
+
+        // read id
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            if (options.truncate_ids)
+            {
+                ranges::copy(stream_view | view::take_until_or_throw(is_blank || is_char<';'>)
```

Comment:
I don't enforce ';' any more because I want to truncate the ids, so as soon as I see a blank, I regard the rest. Should I mention that in a comment and if so, in the implementation notes or just in the line above the code?

Reply:
Okay. Please mention it in the user documentation (implementation notes), because this affects how the user's file is parsed. In the case of the truncate option, consider not only checking for the blank character, but also for whitespace in general. My example above shows how confusing it behaves, if you only consider the blank character.